src/network/validation.rs

```diff
@@ -1,9 +1,9 @@
-use faccess::{AccessMode, PathExt};
 use log::debug;
-use std::path::Path;
+use std::fs::File;
 
 pub fn ns_checks(file: &str) -> std::io::Result<()> {
     debug!("{:?}", "Checking network namespace permissions...");
     // TODO check for FS_MAGIC
-    Path::new(&file).access(AccessMode::EXISTS & AccessMode::WRITE)
+    let _ = !File::open(&file)?.metadata()?.permissions().readonly();
```

Comment:
permisssion.readonly doesn't matter you never use the returned value
The check is performed only by `File::open(&file)?.metadata()?`, The question mark returns the error for you

Reply:
so does it suffice to just stop at getting metadata? Because IIRC, we also needed to check for writability and error otherwise?

Reply:
We can check for writeable but then you have to use the actual return value here and return a error if it is read only

Reply:
ack, let me spend some time to include that. Thanks

Reply:
actually no we cannot check for write permissions, the netns file can be read only it only needs to be a valid fd for the setns call so I think you can stop after metadata

Reply:
ah, I noticed this only now lol. Let me revert my changes.

Reply:
Done, tweaked the debug message as well. PTAL.